### DISCUSSION TITLE
what is the best way to detect if a child route was matched?

### DISCUSSION THREAD

**Discussion by scottbessler** [remix-run/remix#5508] [Q&A]
I have:
*  a route `things` that shows a list of things, and has an outlet to show the detail
* a route `things.$thingid` to show the detail of the thing

I want to hide the "list of things" when (on-mobile && a-thing-is-selected)

Currently, within the `things` handler to determine if "a-thing-is-selected" I'm detecting if it is the "leaf" or "last matched route" using the following:

```
  const lastMatch = useMatches().pop();
  const resolvePath = useResolvedPath(".");
  const isLeaf = resolvePath.pathname === lastMatch?.pathname;
```

Is there a more idiomatic / cleaner / better way to do this?

**Comment by kiliman** [comment]
You can use the `useParams` hook from the parent layout to see if `thingId` is set and render accordingly.
